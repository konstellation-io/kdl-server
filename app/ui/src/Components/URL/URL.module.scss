@import 'Styles/variables';
@import 'Styles/mixins';
@import 'Styles/colors';

.url {
  color: font-color(lowlight);
  cursor: pointer;

  transition: color ease 0.3s;

  &:hover {
    color: palette(highlight);
  }
}
