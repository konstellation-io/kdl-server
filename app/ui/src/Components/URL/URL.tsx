import React, { FC } from 'react';

import cx from 'classnames';
import styles from './URL.module.scss';

type Props = {
  children: string;
  className?: string;
};
const URL: FC<Props> = ({ children, className }) => (
  <div
    className={cx(styles.url, className)}
    onClick={() => window.open(children)}
  >
    {children}
  </div>
);

export default URL;
