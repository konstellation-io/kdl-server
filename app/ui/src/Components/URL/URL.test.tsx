import URL from './URL';
import React from 'react';
import { shallow } from 'enzyme';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<URL children={'foo'} />);
});
describe('URL component', () => {
  it('should render without crashing', () => {
    // Arrange.
    // Act.
    // Assert.
    expect(wrapper).toMatchSnapshot();
  });

  describe('behavior', () => {
    it('should call open function when clicked', () => {
      // Arrange.
      // @ts-ignore
      global.open = jest.fn();
      // Act.
      wrapper.simulate('click');
      // Assert.
      // @ts-ignore
      expect(global.open).toHaveBeenCalled();
    });
  });
});
