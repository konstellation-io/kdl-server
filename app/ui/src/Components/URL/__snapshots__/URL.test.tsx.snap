// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL component should render without crashing 1`] = `
<div
  className="url"
  onClick={[Function]}
>
  foo
</div>
`;
